### PR TITLE
Ny 1uqqq minor changes

### PR DESCRIPTION
**Pull request by nishanty13**
Task 1 & 2

### CODE REVIEW

File: mysite/announcements/admin.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class AnnouncementsAdmin(admin.ModelAdmin):
+    list_display = ('title', 'message', 'date_time_expire')
+    change_form_template = "announcements/announcements_change_form.html"
+
+    def response_change(self, request, obj):
+        if "_send_announcement" in request.POST:
+            group_list = request.POST.getlist('groups')
+            group_users = User.objects.filter(groups__id__in=group_list).distinct()
```
Comment (tarungarg546) [inline mysite/announcements/admin.py]: - Why are we again fetching all columns of user here? Can we just fetch `id`\`username`?
- What is SQL query happening here?

File: mysite/announcements/models.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    updated_at = models.DateTimeField(auto_now=True)
+
+    groups = models.ManyToManyField(Group)
+
+    def __str__(self):
+        return self.title
+
+    def clean(self):
+        # if self.date_time_published <= timezone.now():
+        #     raise ValidationError("The date and time can't be in past")
```
Comment (tarungarg546) [inline mysite/announcements/models.py]: Stray comments :/

It has been pointed out in the previous review too to remove stray comments.

File: mysite/announcements/models.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+from django.utils.encoding import python_2_unicode_compatible
+from django.utils import timezone
+from django.contrib import admin
+from django.contrib.auth.models import User, Group
+
+@python_2_unicode_compatible
+class Announcements(models.Model):
+    title = models.CharField(max_length=32, help_text='This will be shown as the title of notification and announcement')
+    message = models.CharField(max_length=200, help_text='The message you want to convey')
+    # date_time_published = models.DateTimeField('datetime published', help_text='The date and time you want to schedule announcement/notification')
```
Comment (tarungarg546) [inline mysite/announcements/models.py]: Stray comments :/

It has been pointed out in previous review too to remove stray comments.

File: mysite/announcements/views.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.shortcuts import render
+
+# Create your views here.
```
Comment (tarungarg546) [inline mysite/announcements/views.py]: Why did not we remove this file yet? If not in use.

File: mysite/announcements/models.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    groups = models.ManyToManyField(Group)
+
+    def __str__(self):
+        return self.title
+
+    def clean(self):
+        # if self.date_time_published <= timezone.now():
+        #     raise ValidationError("The date and time can't be in past")
+        if self.date_time_expire <= timezone.now():
+            raise ValidationError("The date and time can't be in past")
```
Comment (tarungarg546) [inline mysite/announcements/models.py]: Did we not installed pre-commit yet? If not why?

+ As mentioned in the previous review https://github.com/tarungarg546/annoucements_onboarding/pull/1#discussion_r192972918

File: mysite/announcements/tests.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.test import TestCase
+
+# Create your tests here.
```
Comment (tarungarg546) [inline mysite/announcements/tests.py]: Why did not we remove this file yet? If not in use.

File: .idea/AnnouncementSystem.iml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tarungarg546) [inline .idea/AnnouncementSystem.iml:1]: Please add `.gitignore` on the master branch so that this diff is not visible in this PR.

**Review by tarungarg546 — CHANGES_REQUESTED** [review]

File: mysite/announcements/models.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+class Announcements(models.Model):
+    title = models.CharField(max_length=32, help_text='This will be shown as the title of notification and announcement')
+    message = models.CharField(max_length=200, help_text='The message you want to convey')
+    # date_time_published = models.DateTimeField('datetime published', help_text='The date and time you want to schedule announcement/notification')
+    date_time_expire = models.DateTimeField('datetime expire', help_text='The date and time you want the announcement to expire')
+
+    added_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    groups = models.ManyToManyField(Group)
```
Comment (tarungarg546) [inline mysite/announcements/models.py:17]: I see no migration for this? How would this work on my local?

**Review by tarungarg546 — CHANGES_REQUESTED** [review]

File: mysite/announcements/models.py (hunk 8 above)
Comment (nishanty13) [reply mysite/announcements/models.py:17]: There were migrations but when I squashed them, the new file that was created didn't get commit to git, I will check it out

File: .idea/AnnouncementSystem.iml (hunk 7 above)
Comment (nishanty13) [reply .idea/AnnouncementSystem.iml:1]: Okay

File: mysite/announcements/tests.py (hunk 6 above)
Comment (nishanty13) [reply mysite/announcements/tests.py]: My bad, will do.

File: mysite/announcements/models.py (hunk 5 above)
Comment (nishanty13) [reply mysite/announcements/models.py]: I did install pre-commit but it's not working :/

File: mysite/announcements/views.py (hunk 4 above)
Comment (nishanty13) [reply mysite/announcements/views.py]: Done

File: mysite/announcements/models.py (hunk 3 above)
Comment (nishanty13) [reply mysite/announcements/models.py]: I started working on next task and commented before uploading. Did that count as stray too?

File: mysite/announcements/admin.py (hunk 1 above)
Comment (nishanty13) [reply mysite/announcements/admin.py]: I'll check this out.
Select distinct username from user where id in grouplist

Comment (tarungarg546) [reply mysite/announcements/admin.py]: @nishanty13 
1. Check out what?
2. SQL query is wrong.

File: mysite/announcements/models.py (hunk 3 above)
Comment (tarungarg546) [reply mysite/announcements/models.py]: Yes, counts stray wrt to this task.
For the new task, you should have a new branch.

**Pull request merged**